cmd/virt-launcher/virt-launcher.go

```diff
@@ -496,6 +513,13 @@ func main() {
 			finalShutdownCallback,
 			gracefulShutdownCallback)
 
+		ctx, cancelCtx := context.WithCancel(context.Background())
+		defer cancelCtx()
+
+		if err := runDownwardMetricsVirtioServer(ctx); err != nil {
+			log.Log.Reason(err).Error("failed to start the DownwardMetrics server")
+			cancelCtx()
```

Comment:
You should be able to drop this call to `cancelCtx()`.

Reply:
Should this lead to termination of the Pod / VM? How can we better highlight that something has failed?

Reply:
The server may fail to start just for two reasons:

1. launcher-sock is not found.
2. Env variables are not found.

Once we have reached `runDownwardMetricsVirtioServer` call point, the first reason is just impossible since some line above we can find `markReady()` function, which it renames the file from `launcher-init-sock` to `launcher-sock` or panics. The second reason is also very unlikely, it would mean that the render launcher pod template logic has been modified.

To avoid the second reason, I've included a new unit test. In this way, we can be sure that it is not modified unintentionally.

Reply:
Let's keep it for some time, just in case others reviews have the same concern.

Reply:
IMO it gives a false sense of security. Code and comments can diverge, comments should not state that something can never happen.

Reply:
You can still drop this cancel

Reply:
Yeah true, sorry, missed it again :sweat_smile:

Reply:
Dropped.